server/internal/mv/organization.go

```diff
@@ -37,6 +37,14 @@ func DescribeOrganization(ctx context.Context, logger *slog.Logger, orgRepo *org
 	// Otherwise, the source of truth for account type is the Polar customer state
 	if customerTier != nil {
 		org.GramAccountType = string(*customerTier)
+		if org.GramAccountType != string(*customerTier) {
```

Comment:
It concerns me slightly that this logic is reliant on the enterprise early return above. Is it worth adding a check here that the org is not enterprise before overwriting?